### PR TITLE
Fix wrong output size in SplineC2ROMPTarget::mw_evaluateVGLandDetRatioGrads

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
There is bug when using offload real build.
This caused writing to undesired memory locations and wrong numbers (crazy high variance/energy >>1)

## What type(s) of changes does this code introduce?
- Bugfix
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp [hunk 1]
```diff
@@ -942,31 +958,32 @@ void SplineC2ROMPTarget<ST>::mw_evaluateVGLandDetRatioGrads(const RefVectorWithL
...
           spline2offload::evaluate_vgh_impl_v2(spline_ptr, ix, iy, iz, a, b, c, da, db, dc, d2a, d2b, d2c,
                                                offload_scratch_iw_ptr + first,
-                                               offload_scratch_iw_ptr + padded_size + first,
-                                               offload_scratch_iw_ptr + padded_size * 4 + first, padded_size, first,
-                                               index);
+                                               offload_scratch_iw_ptr + spline_padded_size + first,
+                                               offload_scratch_iw_ptr + spline_padded_size * 4 + first,
+                                               spline_padded_size, first, index);
         const size_t first_cplx = first / 2;
-        const size_t last_cplx  = omptarget::min(last / 2, orb_size);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp:949]: The `min()` here is wrong. orb_size refers to output real orbitals.

File: src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp [hunk 2]
```diff
@@ -942,31 +958,32 @@ void SplineC2ROMPTarget<ST>::mw_evaluateVGLandDetRatioGrads(const RefVectorWithL
...
 
-        ValueType* restrict out_phi    = phi_vgl_ptr + iw * orb_size;
+        ValueType* restrict out_phi    = phi_vgl_ptr + iw * requested_orb_size;
         ValueType* restrict out_dphi_x = out_phi + phi_vgl_stride;
         ValueType* restrict out_dphi_y = out_dphi_x + phi_vgl_stride;
         ValueType* restrict out_dphi_z = out_dphi_y + phi_vgl_stride;
         ValueType* restrict out_d2phi  = out_dphi_z + phi_vgl_stride;
 
         const size_t first_real = first_cplx + omptarget::min(nComplexBands_local, first_cplx);
-        const size_t last_real  = last_cplx + omptarget::min(nComplexBands_local, last_cplx);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp:969]: `last_real` can be larger than `requested_orb_size` and caused overflow.

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — COMMENTED** [review]
Thanks Ye. We discussed this a bit last week, but so we have a record here: why didn't asan catch this?

**Comment by ye-luo** [comment]
> Thanks Ye. We discussed this a bit last week, but so we have a record here: why didn't asan catch this?

Because we don't have a test case exposing this bug. The added test triggers asan failure when I backported it to develop.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
I also noticed that asan runs on the github actions without QMC_DATA. We probably need to update the image to include NiO a4(S1) and a16(S4) h5 files.

**Review by prckent — APPROVED** [review]
LGTM. Coverage would be higher but for #4409 .

We (I) should mention in the release notes that QMC_DATA and these NiO files are encouraged.

**Pull request merged**